src/chunk.c

```diff
@@ -1322,24 +1325,35 @@ ts_chunk_find_or_create_without_cuts(const Hypertable *ht, Hypercube *hc, const
  * for share. NULL if not found.
  */
 Chunk *
-ts_chunk_find_for_point(const Hypertable *ht, const Point *p)
+ts_chunk_find_for_point(const Hypertable *ht, const Point *p, LOCKMODE lockmode)
 {
-	int chunk_id = chunk_point_find_chunk_id(ht, p);
+	int32 chunk_id = ts_chunk_point_find_chunk_id(ht, p, false);
+
 	if (chunk_id == INVALID_CHUNK_ID)
-	{
 		return NULL;
-	}
+
+	ScanTupLock slice_lock = {
+		.lockmode = LockTupleKeyShare,
```

Comment:
Unless I am mistaken, this will allow the `range_start` and `range_end` to be modified while the lock is held. Is this intentional? If we just remove existing dimensions this is fine since it will block DELETE of the tuple, but if we can _update_ dimension slices, the situation is different.